sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ObjectShouldBeInitializedCorrectlyBase.cs

```diff
@@ -0,0 +1,210 @@
+﻿/*
+ * SonarAnalyzer for .NET
+ * Copyright (C) 2015-2018 SonarSource SA
+ * mailto: contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using SonarAnalyzer.Helpers;
+
+namespace SonarAnalyzer.Rules
+{
+    public abstract class ObjectShouldBeInitializedCorrectlyBase : SonarDiagnosticAnalyzer
+    {
+        protected abstract DiagnosticDescriptor Rule { get; }
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        /// <summary>
+        /// Gets the KnownType representing the type on which instances the rule will raise issues on.
+        /// </summary>
+        internal abstract KnownType TrackedType { get; }
+
+        /// <summary>
+        /// Gets the name of the property that has to be set with allowed value in
+        /// order for the object to be initialized correctly.
+        /// </summary>
+        protected abstract string TrackedPropertyName { get; }
+
+        /// <summary>
+        /// Tests if the provided <paramref name="constantValue"/> is equal to allowed value.
+        /// </summary>
+        /// <returns>True when <paramref name="constantValue"/> is an allowed value, otherwise false.</returns>
+        protected abstract bool IsAllowedValue(object constantValue);
+
+        protected override void Initialize(SonarAnalysisContext context)
+        {
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    if (c.Node.GetLocation().GetLineNumberToReport() == 8)
+                    {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Either remove or fill this block of code. ([csharpsquid:S108](https://sonarcloud.io/organizations/sonarsource/rules#rule_key=csharpsquid%3AS108))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=sonaranalyzer-dotnet&pullRequest=1351&issues=AWMBR9oHmtQEh111DpXq&open=AWMBR9oHmtQEh111DpXq)

[ISSUE_KEY]: AWMBR9oHmtQEh111DpXq